test/libsolidity/SemanticTest.cpp

```diff
@@ -151,8 +152,38 @@ map<string, Builtin> SemanticTest::makeBuiltins() const
 			[this](FunctionCall const& _call) -> optional<bytes>
 			{
 				soltestAssert(_call.arguments.parameters.empty(), "No arguments expected.");
-				  return toBigEndian(u256(storageEmpty(m_contractAddress) ? 1 : 0));
+				return toBigEndian(u256(storageEmpty(m_contractAddress) ? 1 : 0));
 		 	}
+		},
+		{
+			"bytecode",
+			[this](FunctionCall const& _call) -> optional<bytes>
+			{
+				bool result{false};
+				optional<regex> regexPattern;
+				soltestAssert(_call.arguments.parameters.size() == 1, "Bytecode pattern string expected.");
+				string pattern{_call.arguments.parameters.at(0).rawString};
+				bytes bytecode = ExecutionFramework::bytecode();
```

Comment:
As discussed in https://github.com/ethereum/solidity/pull/11424/files#r636862608, this only works if used after the `constructor()` line. Is there a way to enforce that?

If not, I would just leave a comment here in case anyone runs into it.

Reply:
Maybe we could use `_call.signature`? Does constructor signature always start with `constructor(` or is it something special to avoid collisions with a function called `constructor()`?

Reply:
We could use a side-effects hook to keep track on what call is executed. With this we could easily check what the previous call actually was. I will rebase this PR on top of #11048.